File: src/main/java/no/fintlabs/request/RequestFintEventRepository.java
package no.fintlabs.request;

import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import java.time.Instant;
import java.time.temporal.ChronoUnit;

@Slf4j
@Service
public class RequestFintEventRepository {

    private RequestFintEventJpaRepository requestFintEventJpaRepository;

    public RequestFintEventRepository(RequestFintEventJpaRepository requestFintEventJpaRepository) {
        this.requestFintEventJpaRepository = requestFintEventJpaRepository;
    }

    @Scheduled(cron = "${event.database.cleanupTime}")
    private void removeEventsOlderThanTwoWeeks() {
        long days = Instant.now().minus(30, ChronoUnit.DAYS).toEpochMilli();
        try {
            requestFintEventJpaRepository.deleteRowsOlderThan(days);
        } catch (Exception e) {
            log.error(e.getMessage(), e);
        }
    }

}

File: src/main/kotlin/no/fintlabs/event/request/RequestFintEventRepository.kt
package no.fintlabs.event.request

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service
import java.time.Instant
import java.time.temporal.ChronoUnit

@Service
class RequestFintEventRepository(
    private val requestFintEventJpaRepository: RequestFintEventJpaRepository
) {

    private val log = LoggerFactory.getLogger(RequestFintEventRepository::class.java)

    @Scheduled(cron = "\${event.database.cleanupTime}")
    private fun removeEventsOlderThanTwoWeeks() {
        val days = Instant.now().minus(14, ChronoUnit.DAYS).toEpochMilli()
        try {
            requestFintEventJpaRepository.deleteRowsOlderThan(days)
        } catch (e: Exception) {
            log.error(e.message, e)
        }
    }

}
